src/config.rs

```diff
@@ -1275,6 +1278,138 @@ pub fn check_and_persist_critical_config(config: &TiKvConfig) -> Result<(), Stri
     Ok(())
 }
 
+/**
+|total_size|key1_size|key1|value1_size|value1|key2_size|key2|value2_size|value2|...
+all size is four bytes.
+existing_value : existing_val
+operand_list   : operands
+new_value      : result
+Note: little endian Coding
+*/
+fn update_merge(_: &[u8], existing_val: Option<&[u8]>, operands: &mut MergeOperands) -> Vec<u8> {
+    let mut result: Vec<u8> = Vec::with_capacity(operands.size_hint().0);
+    let arr : [u8;4] = [0, 0, 0, 0];
+    result.extend_from_slice(&arr[..]);
+
+    let mut map_index = HashMap::new();
+
+    let mut total_size : u32 = 0;
+    total_size += 4;
+
+    let mut v = Vec::new();
+    for op in operands {
+        v.push(op);
+    }
+    let len = v.len();
+
+    for i in 0..len {
+        let op_value : &[u8] = v.get( len-1-i ).unwrap();
+        let mut begin = 0;
+        begin += 4 ;
+
+        let op_len = op_value.len();
+        while begin < op_len {
+
+            //decode key size
+            let first = begin;
+            let end = begin + 4 ;
+            let vv = &op_value[begin..end];
+            let ptr: *const u8  = vv.as_ptr();
+            let ptr: *const u32 = ptr as *const u32;
+            let key_size = unsafe{ *ptr };
+
+            //decode key
+            begin = end;
+            let end = begin + key_size as usize;
+            let key = &op_value[begin..end];  
+
+
+            //decode value size
+            begin = end;
+            let end = begin + 4;
+            let vv = &op_value[begin..end];
+            let ptr: *const u8  = vv.as_ptr();
+            let ptr: *const u32 = ptr as *const u32;
+            let value_size = unsafe{ *ptr };
+
+            //decode value
+            begin = end;
+            let end = begin + value_size as usize;
+            //let value = &op_value[begin..end];
+
+            if let None = map_index.get(&key) {
+                map_index.entry(key).or_insert(1);
+
+                if value_size != 0 {
+                    result.extend_from_slice(&op_value[first..end]);
+                    total_size += 8 + key_size + value_size;
+                }
+
+            }
+            begin = end;
+        }
+    }
+
+    if let Some(existing_value) = existing_val {
+        let mut begin = 0;
+        let exist_total_size = existing_value.len();
+
+        begin += 4 ;
+
+        while begin < exist_total_size {
+            //decode key size
```

Comment:
Please place a space between `//` and following text, like:
```rust
// Decode key size.
```
The same to other comments here